src/cdk/table/table.ts

```diff
@@ -515,6 +542,88 @@ export class CdkTable<T> implements AfterContentChecked, CollectionViewer, OnDes
     this._footerRowDefChanged = true;
   }
 
+  /**
+   * Updates the header sticky styles. First resets all applied styles with respect to the cells
+   * sticking to the top, left, and right. Then, evaluating which cells need to be stuck to the top.
+   * Then, adding sticky left and sticky right according to the column definitions for each cell
+   * in each row. This is automatically called when the header row changes its displayed set of
+   * columns, if its sticky input changes, or when any column definition changes its sticky input.
+   * May be called manually for cases where the cell content changes outside of these events.
+   */
+  updateStickyHeaderRowStyles() {
+    const headerRows = this._getRenderedRows(this._headerRowOutlet);
+    this._stickyStyler.clearStickyPositioningStyles(headerRows, ['top']);
+
+    const stickyStates = this._headerRowDefs.map(def => def.sticky);
+    this._stickyStyler.stickRows(headerRows, stickyStates, 'top');
+
+    // Reset the dirty state of the sticky input change since it has been used.
+    this._headerRowDefs.forEach(def => def.resetStickyChanged());
+  }
+
+  /**
+   * Updates the footer sticky styles. First resets all applied styles with respect to the cells
+   * sticking to the bottom, left, and right. Then, evaluating which cells need to be stuck to the
+   * bottom. Then, adding sticky left and sticky right according to the column definitions for each
+   * cell in each row. This is automatically called when the footer row changes its displayed set of
+   * columns, if its sticky input changes, or when any column definition changes its sticky input.
+   * May be called manually for cases where the cell content changes outside of these events.
+   */
+  updateStickyFooterRowStyles() {
+    const footerRows = this._getRenderedRows(this._footerRowOutlet);
+    this._stickyStyler.clearStickyPositioningStyles(footerRows, ['bottom']);
+
+    const stickyStates = this._footerRowDefs.map(def => def.sticky);
+    this._stickyStyler.stickRows(footerRows, stickyStates, 'bottom');
+    this._stickyStyler.updateStickyFooterContainer(this._elementRef.nativeElement, stickyStates);
+
+    // Reset the dirty state of the sticky input change since it has been used.
+    this._footerRowDefs.forEach(def => def.resetStickyChanged());
+  }
+
+  /**
+   * Updates the column sticky styles. First resets all applied styles with respect to the cells
+   * sticking to the left and right. Then sticky styles are added for the left and right according
+   * to the column definitions for each cell in each row. This is automatically called when
+   * the data source provides a new set of data or when a column definition changes its sticky
+   * input. May be called manually for cases where the cell content changes outside of these events.
+   */
+  updateStickyColumnStyles() {
```

Comment:
You have another method with the same name that's internal; they should have different names if they're doing different things

Reply:
Thanks, didn't catch that. Kept this as update (it removes/adds) and the internal one as add (it only adds)